### PR TITLE
Deduplicated linking

### PR DESCRIPTION
**Pull request by DeNeutoy**
Moves from including each alias per concept to just having a unique set of words, which each have a mapping back to concepts that they are aliases for. Also adds some code around thresholding NN results by similarity.

### CODE REVIEW

File: scripts/linking.py [hunk 1]
```diff
@@ -71,19 +71,24 @@ class CandidateGenerator:
         The vectorizer used to encode mentions.
     ann_concept_id_list: List[str]
```
Comment (danielkingai2) [inline scripts/linking.py]: nit: change parameter name (and description if needed)

Comment (DeNeutoy) [reply scripts/linking.py]: Done, thanks, FYI having correct docstrings is not a nit in my book

File: scripts/linking.py [hunk 2]
```diff
@@ -246,15 +235,15 @@ def create_tfidf_ann_index(model_path: str, umls_concept_list: List[str]) -> Non
 def load_tfidf_ann_index(model_path: str):
     # `S` for Search. This controls performance at query time. Maximum recommended value is 2000.
     # It makes the query slow without significant gain in recall.
-    efS = 100
+    efS = 1000
```
Comment (ibeltagy) [inline scripts/linking.py]: Just curious, why increasing `efS` back again ?

Comment (DeNeutoy) [reply scripts/linking.py]: Oh sorry, I was just playing around. Once we move it into the main codebase we can be more strict about stuff.

**Pull request merged**